packages/story-editor/src/components/canvas/useInsertElement.ts

```diff
@@ -60,13 +63,17 @@ function useInsertElement() {
    * @param {boolean} insertAsBackground Whether to insert the element as a background element.
    */
   const insertElement = useCallback(
-    (type, props, insertAsBackground = false) => {
+    (
+      type: ElementType,
+      props: ElementWithPageId,
+      insertAsBackground = false
+    ) => {
       setZoomSetting(ZoomSetting.Fit);
-      const element = createElementForCanvas(type, props);
-      const { id, resource, pageId } = element;
+      const element = createElementForCanvas(type, props) as ElementWithPageId;
+      const { id, pageId } = element;
       addElement({ element, pageId });
```

Comment:
Also, aren't we removing the `pageId` from the element again? Or does that happen in the reducer?